components/content-service/pkg/storage/s3.go

```diff
@@ -289,33 +290,36 @@ func (s3st *s3Storage) DownloadSnapshot(ctx context.Context, destination string,
 	return s3st.download(ctx, destination, name, mappings)
 }
 
+// download object using s5cmd (prior to which we used aws sdk)
 func (s3st *s3Storage) download(ctx context.Context, destination string, obj string, mappings []archive.IDMapping) (found bool, err error) {
-	downloader := s3manager.NewDownloader(s3st.client, func(d *s3manager.Downloader) {
-		d.Concurrency = defaultCopyConcurrency
-		d.PartSize = defaultPartSize * megabytes
-		d.BufferProvider = s3manager.NewPooledBufferedWriterReadFromProvider(25 * megabytes)
-	})
-
-	s3File, err := os.CreateTemp("", "temporal-s3-file")
+	tempFile, err := os.CreateTemp("", "temporal-s3-file")
 	if err != nil {
 		return true, xerrors.Errorf("creating temporal file: %s", err.Error())
 	}
-	defer os.Remove(s3File.Name())
+	tempFile.Close()
 
-	_, err = downloader.Download(ctx, s3File, &s3.GetObjectInput{
-		Bucket: aws.String(s3st.Config.Bucket),
-		Key:    aws.String(obj),
-	})
+	args := []string{
+		"--numworkers", "20",
```

Comment:
This should not be --concurrency instead of --numworkers? (from the docs)

Reply:
Oh great [catch](https://github.com/peak/s5cmd#concurrency)! That means it should go even faster than we saw in [this test](https://linear.app/gitpod/issue/ENG-884/[content-service]-improve-aws-s3-speed).

Reply:
[numworkers](https://github.com/peak/s5cmd#numworkers) would be good for operating on (read, cp, select) many files at once (one worker per file). The default is 256. I don't think it'll help with a single download of a file.

From their docs:

> If you have a few, large files to download, setting --numworkers to a very high value will not affect download speed. In this scenario setting --concurrency to a higher value may have a better impact on the download speed.

[concurrency](https://github.com/peak/s5cmd#concurrency) is for controlling the number of parts for a `cp` operation on a single file. default is 5. In my testing, I set it to 10. I'll try with 20, to see how much of an improvement there is for posterity.

Reply:
Interesting, in my test notes, I used `--num-workers` 20, instead of `concurrency`. So, I'll run a few tests, with `concurrency` at 10 and 20.

Reply:
`--concurreny` was correct, GTG. :smile: